src/System.Private.Uri/tests/FunctionalTests/UriTests.cs

```diff
@@ -892,5 +892,63 @@ public static void Uri_ExtremelyLongScheme_ThrowsUriFormatException()
             string largeString = new string('a', 1_000_000) + ":"; // 2MB is large enough to cause a stack overflow if we stackalloc the scheme buffer.
             Assert.Throws<UriFormatException>(() => new Uri(largeString));
         }
+
+        [Fact]
+        public static void Uri_HostTrailingSpaces_SpacesTrimmed()
+        {
+            string host = "www.contoso.com";
+            Uri u = new Uri($"http://{host}     ");
+
+            Assert.Equal($"http://{host}/", u.AbsoluteUri);
+            Assert.Equal(host, u.Host);
+        }
+
+        [Theory]
+        [InlineData("1234")]
+        [InlineData("01234")]
+        [InlineData("12340")]
+        [InlineData("012340")]
+        [InlineData("99")]
+        [InlineData("09")]
+        [InlineData("90")]
+        [InlineData("0")]
+        [InlineData("000")]
```

Comment:
What happens if the port is larger than is allowed, e.g. ushort.MaxValue + 1 ?

Reply:
Ideally we should be throwing a UriFormatException. I'll verify that behavior with a test.